File: backend/app/demo_scenario.py
"""Deterministic demo spawn plan for interview-safe Run Demo."""

from __future__ import annotations

from typing import TypedDict


class SpawnPlanEntry(TypedDict):
    id: str
    model: str
    paint: str
    battery: float
    target_battery: float
    spot_id: str
    spawn_at_tick: int
    departure_offset: int
    requested_kwh: float


# Only one scripted arrival during the demo story; overnight fleet is pre-seeded.
DEMO_VEHICLE_SPAWN_PLAN: list[SpawnPlanEntry] = [
    {
        "id": "EV-4466",
        "model": "Nissan Ariya",
        "paint": "silver",
        "battery": 29,
        "target_battery": 75,
        "spot_id": "A5",
        "spawn_at_tick": 10,
        "departure_offset": 12000,
        "requested_kwh": 22.0,
    },
]

# Optional slow post-plan arrivals: mostly overnight.
OVERNIGHT_PROBABILITY = 0.85
OVERNIGHT_DEPARTURE_OFFSET_MIN = 8000
OVERNIGHT_DEPARTURE_OFFSET_MAX = 16000
SHORT_STAY_DEPARTURE_OFFSET_MIN = 2500
SHORT_STAY_DEPARTURE_OFFSET_MAX = 5000

File: frontend/src/lib/ops/demoScenario.ts
import type { VehiclePaint } from "../types";

export interface DemoSpawnPlanEntry {
  id: string;
  model: string;
  paint: VehiclePaint;
  battery: number;
  targetBattery: number;
  spotId: string;
  spawnAtTick: number;
  departureOffset: number;
  requestedKwh: number;
  parkingCategory: "overnight" | "short_stay";
}

/** Only one scripted arrival; overnight fleet is pre-seeded. */
export const DEMO_VEHICLE_SPAWN_PLAN: DemoSpawnPlanEntry[] = [
  {
    id: "EV-4466",
    model: "Nissan Ariya",
    paint: "silver",
    battery: 29,
    targetBattery: 75,
    spotId: "A5",
    spawnAtTick: 10,
    departureOffset: 12000,
    requestedKwh: 22,
    parkingCategory: "overnight",
  },
];

export const OVERNIGHT_PROBABILITY = 0.85;
export const DEMO_TARGET_VEHICLE_ID = "EV-4466";
